config/routes.rb

```diff
@@ -28,6 +32,11 @@
   get "/register", to: "users#new"
   get "/logout", to: "authentications#destroy"
 
+  get '/password/reset', to: 'password_resets#new'
+  post '/password/reset', to: 'password_resets#create'
+  get '/password/reset/edit', to: 'password_resets#edit'
+  patch '/password/reset/edit', to: 'password_resets#update'
```

Comment:
ah yeah I was following some instructions and I forgot to clean that part up. I added `resources :password_resets, only: [:new, :create, :edit, :update]` and the resulting routes are:

```
                         password_resets POST   /password_resets(.:format)                                                                        password_resets#create
                      new_password_reset GET    /password_resets/new(.:format)                                                                    password_resets#new
                     edit_password_reset GET    /password_resets/:id/edit(.:format)                                                               password_resets#edit
                          password_reset PATCH  /password_resets/:id(.:format)                                                                    password_resets#update
                                         PUT    /password_resets/:id(.:format)                                                                    password_resets#update
```

so the added :id url parameters don't work with this, but I think I can add some different routes to make more sense of this

Reply:
@jlvallelonga just drop the 's' so "resources" becomes `resource`. I didn't catch that.

Reply:
I ended up adding this:
```
  resources :password_resets, only: [:new, :create]
  resource :password, only: [:edit, :update]
```

which gives us the following routes:
```
                         password_resets POST   /password_resets(.:format)                                                                        password_resets#create
                      new_password_reset GET    /password_resets/new(.:format)                                                                    password_resets#new
                           edit_password GET    /password/edit(.:format)                                                                          passwords#edit
                                password PATCH  /password(.:format)                                                                               passwords#update
                                         PUT    /password(.:format)                                                                               passwords#update
```

I think it makes sense to split these into two different RESTful resources: password_resets and passwords. So I created another controller for passwords. Creating a password_reset creates the reset token and sends the email and then you can edit and update the password for the user which is a different kind of resource. It felt wrong to update a user's password by editing and updating the "password reset"

Reply:
Also, I thought about nesting this under users/:token/password, but it felt weird to have the really long token as a url parameter instead of a query parameter. i.e. `users/<really long token>/password` was weird so I went with `/password?token=<really long token>`

Reply:
I agree with that. I’ve come to avoid nested resources accept in rare cases where it really adds something. That is mostly only in the case where I have one resource unnested and another version nested.

Reply:
I also agree it makes sense to have separate resources. Adding more resources to try and stick to CRUD always seems to be worth it.

But on password, I wonder if it should just be under person/edit but that’s a bit more complicated with the nested person > user > password credential hierarchy so it’s probably worth the dedicated password route, like you did.

Actually, I might lean towards naming the route password_credential since that’s the model it’s related to, but password seems fine too. It’s a bit like when I added “signup” for users/create just because it’s a prettier route.

Reply:
I like the password_credential route matching the model so I made that change too